2009-02-13  Michael Tautschnig  <mt@debian.org>

	* examples/simple: Made all scripts idempotent and softupdate-capable
		(closes: #481871)
Index: trunk/examples/simple/files/boot/grub/menu.lst/postinst
===================================================================
--- trunk.orig/examples/simple/files/boot/grub/menu.lst/postinst
+++ trunk/examples/simple/files/boot/grub/menu.lst/postinst	
@@ -4,8 +4,12 @@
 
 set -a
 
-# during softupdate use this file
-[ -r $target/var/log/fai/disk_var.sh ] && . $target/var/log/fai/disk_var.sh
+if [ -r $LOGDIR/disk_var.sh ] ; then
+  . $LOGDIR/disk_var.sh
+else
+  echo "disk_var.sh not found!"
+  exit 0
+fi
 
 # if class NOMBR is defined, write boot loader into root partition, not into mbr
 ifclass NOMBR && BOOT_DEVICE=$BOOT_PARTITION
Index: trunk/examples/simple/scripts/FAIBASE/30-interface
===================================================================
--- trunk.orig/examples/simple/scripts/FAIBASE/30-interface
+++ trunk/examples/simple/scripts/FAIBASE/30-interface	
@@ -2,7 +2,7 @@
 
 error=0 ; trap "error=$((error|1))" ERR
 
-if ifclass DHCPC
+if ifclass DHCPC && [ $FAI_ACTION != "softupdate" ]
 then
     cat > $target/etc/network/interfaces <<-EOF
 	# generated by FAI
@@ -23,11 +23,11 @@
 EOF
     [ -n "$NETWORK" ] && echo "localnet $NETWORK" > $target/etc/networks
     [ -s /etc/resolv.conf ] && cp -p /etc/resolv.conf $target/etc
-    fcopy -i /etc/resolv.conf
 fi
 
 # here fcopy is mostly used, when installing a client for running in a
 # different subnet than during the installation  
+fcopy -i /etc/resolv.conf
 fcopy -iM /etc/network/interfaces /etc/networks
 
 exit $error
Index: trunk/examples/simple/scripts/GRUB/10-setup
===================================================================
--- trunk.orig/examples/simple/scripts/GRUB/10-setup
+++ trunk/examples/simple/scripts/GRUB/10-setup	
@@ -2,7 +2,7 @@
 
 error=0 ; trap "error=$((error|1))" ERR
 
-fcopy -Uv /boot/grub/menu.lst
+fcopy -v /boot/grub/menu.lst
 
 exit $error
 
